### DISCUSSION TITLE
想做一个基于PAG模板的视频生成SaaS平台，想问下建议用哪个端的SDK好一些？

### DISCUSSION THREAD

**Discussion by weizmann** [Tencent/libpag#1579] [General] [answered]
公司里有一些视频包装都是用固定的AE特效，想把这些AE特效用PAG封装一下

大家往模板里面做填空题，（Q1）想问下如何这个需求场景，PAG可以实现么？

我们的视频生成一般是后端服务器逐帧渲染。（Q2）想问下这个需求，如果用PAG的话，应该用哪个端的SDK更合适一些？web or Linux？

另外，(Q3)这是一个PAG常见的应用场景么？大家有走通的case么？

**Comment by kevingpqi123** [comment, marked as answer]
这些都是最基础的能力，可以了解下微信中的游戏战报、腾讯 AMS 的视频广告，采用的都是 pag 的技术方案，支持对于一段视频添加多个特效，且特效中的文本和占位图都是支持二次编辑的，这类方案基本上都是通过服务端渲染的方案， web 端的性能限制太大了，目前这部分技术很成熟
